src/__/arrays/dropWhile.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace arrays;
+
+/**
+ * In order to return either an array or \Generator from dropWhile, we split out
+ * the logic or handling arrays and other other traversables.
+ *
+ * @internal
+ *
+ * @param \Traversable  $input
+ * @param \Closure      $comparison
+ *
+ * @return \Generator
+ */
+function dropIterableWhile($input, $comparison)
+{
+    $dropping = true;
+    $iterator = \__::getIterator($input);
+
+    foreach ($iterator as $item) {
+        if ($dropping && $comparison($item)) {
```

Comment:
When would `$dropping` ever be false here?

Reply:
As written, no it won't. Good catch! I went ahead and updated the code to prevent comparisons once dropping has finished. I also added a test specifically for this scenario.